From e067449042759039ef1f79e9abfc6dfb706885e6 Mon Sep 17 00:00:00 2001
From: Xulin Sun <xulin.sun@windriver.com>
Date: Thu, 29 Jan 2015 15:43:30 +0800
Subject: [PATCH 4/4] gianfar: use __raw_get_cpu_var to replace __get_cpu_var

__get_cpu_var would create below call trace in preemptible context situation,
simply use __raw_get_cpu_var to replace __get_cpu_var in preemptible context
situation to avoid the call trace.

BUG: using smp_processor_id() in preemptible [00000000] code: ifconfig/281
caller is gfar_new_skb+0x34/0x280
CPU: 0 PID: 281 Comm: ifconfig Not tainted 3.10.62-ltsi-WR6.0.0.0_cgl #1
Call Trace:
[c7d01c80] [c0008460] show_stack+0xfc/0x1c8 (unreliable)
[c7d01cd0] [c04b7238] debug_smp_processor_id+0xe4/0xf4
[c7d01cf0] [c05eb2d4] gfar_new_skb+0x34/0x280
[c7d01d40] [c05eb678] gfar_init_bds+0x158/0x1c4
[c7d01d80] [c05eba7c] startup_gfar+0x398/0x50c
[c7d01dc0] [c075aa98] __dev_open+0x11c/0x18c
[c7d01de0] [c075ad9c] __dev_change_flags+0xac/0x178
[c7d01e00] [c075af34] dev_change_flags+0x28/0x74
[c7d01e20] [c07cb248] devinet_ioctl+0x5bc/0x924
[c7d01e80] [c073bbb8] sock_ioctl+0x324/0x3bc
[c7d01eb0] [c01a28c4] do_vfs_ioctl+0x498/0x70c
[c7d01f10] [c01a2bc8] SyS_ioctl+0x90/0x9c
[c7d01f40] [c0010f04] ret_from_syscall+0x0/0x3c

Signed-off-by: Xulin Sun <xulin.sun@windriver.com>
---
 drivers/net/ethernet/freescale/gianfar.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/drivers/net/ethernet/freescale/gianfar.c b/drivers/net/ethernet/freescale/gianfar.c
index 16cccef..088e64d 100644
--- a/drivers/net/ethernet/freescale/gianfar.c
+++ b/drivers/net/ethernet/freescale/gianfar.c
@@ -3081,7 +3081,7 @@ struct sk_buff *gfar_new_skb(struct net_device *dev, dma_addr_t *bufaddr)
 	dma_addr_t addr;
 
 	if (likely(priv->rx_buffer_size <= DEFAULT_RX_BUFFER_SIZE)) {
-		struct sk_buff_head *h = &__get_cpu_var(skb_recycle_list);
+		struct sk_buff_head *h = &__raw_get_cpu_var(skb_recycle_list);
 		skb = __skb_dequeue(h);
 	}
 
-- 
1.7.5.4

